src/mono/mono/mini/intrinsics.c

```diff
@@ -488,6 +488,125 @@ emit_unsafe_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatu
 		EMIT_NEW_BIALU (cfg, ins, OP_COMPARE, -1, args [0]->dreg, args [1]->dreg);
 		EMIT_NEW_UNALU (cfg, ins, OP_PCEQ, dreg, -1);
 		return ins;
+	} else if (!strcmp (cmethod->name, "BitCast")) {
+		g_assert (ctx);
+		g_assert (ctx->method_inst);
+		g_assert (ctx->method_inst->type_argc == 2);
+		g_assert (fsig->param_count == 1);
+
+		MonoType *tfrom = ctx->method_inst->type_argv [0];
+		if (mini_is_gsharedvt_variable_type (tfrom)) {
+			// FIXME: Handle gsharevt
+			return NULL;
+		}
+
+		MonoTypeEnum tfrom_type = tfrom->type;
+		MonoClass *tfrom_klass = mono_class_from_mono_type_internal (tfrom);
+		if (!m_class_is_valuetype (tfrom_klass)) {
+			return NULL;
+		}
```

Comment:
I think you want `if (mono_type_is_reference(tfrom_type)) return NULL`.  (although i'm not certain about generic params)

Reply:
It looks correct, checking the underlying implementation, have updated to use it. 👍